### ISSUE TITLE
New lint: Use `def` instead of lambda expression for bare assignments and calls

### ISSUE THREAD

**Issue by jpy-git** [PyCQA/pylint#5976]
### Current problem

Currently these uses of lambda expressions don't raise any lint.
A)
```python
a = lambda x: x
```
(^ Same as E731 in flake8)

B)
```python
(lambda x: x)("hello world")
```
(not flagged by flake8 or pylint)

### Desired solution

A)
```python
a = lambda x: x
```
raises lint to tell user to create a normal function instead of assigning a lambda.
Should be replaced with
```python
def a(x)
    return x
```

B)
```python
y = (lambda x: x.upper())("hello world")
```
raises lint telling user to just execute the lambda body since there's no need to define a function or lambda expression to compute something once.
Should be replaced with
```python
x = "hello world"
y = x.upper()
```

**Comment by jpy-git** [comment]
N.B. there is `W0108: unnecessary-lambda` but that seems to focussed on cases where you could replace the lambda with it's contents if it just contains a simple callable e.g. `lambda: func()` -> `func`.

The suggestion in this issue is about not using lambdas in inappropriate places

**Comment by jacobtylerwalls** [comment]
Just to be clear, you're proposing only assignments and calls, not arguments to calls, right?

I don't think this should emit anything:
```python
results.sort(key=lambda x: x[1])
```

**Comment by jpy-git** [comment]
yes using it a lambda expression in an argument is completely fine 😄
